tutorials/language/tensorize.py

```diff
@@ -0,0 +1,286 @@
+# If the strides can be inferred
+# - in this case, TVM knows tensor B is compact thus the strides are :code:`[L, 1]` -
+# such placeholder can be put to let TVM automatically bind the inferred value for us.
+#
+gemv = intrin_gemv(factor, L)
+s[C].tensorize(yi, gemv)
+print(tvm.lower(s, [A, B, C], simple_mode=True))
+
+######################################################################
+# By tensorizing over :code:`yi`, the inner most two loops are
+# now replaced by the intrinsic function we defined before.
+# In order to build and run the module, let's define the external function :code:`gemv_update`,
+# it is a naive implementation of GEMV, just for demonstration.
+#
+def gemv_impl():
+    cc_code = """
+      extern "C" int gemv_update(float *cc, float *aa, float *bb, int m, int l, int stride) {
+        for (int i = 0; i < m; ++i) {
+            for (int j = 0; j < l; ++j) {
+                cc[i] += aa[j] * bb[i * stride + j];
+            }
+        }
+        return 0;
+      }
+    """
+    from tvm.contrib import util, clang
+    temp = util.tempdir()
+    ll_path = temp.relpath("temp.ll")
+    # Create LLVM ir from c source code
+    ll_code = clang.create_llvm(cc_code, output=ll_path)
+    return ll_code
+
+######################################################################
+# Now we leverage the pragma attribute :code:`import_llvm` to import llvm asm inline.
+# The importing needs to happen before the tensorized GEMV being executed.
+#
+s[C].pragma(x, "import_llvm", gemv_impl())
+func = tvm.build(s, [A, B, C], target="llvm", name="gemv")
+
+from topi.util import get_const_tuple
+dtype = A.dtype
+ctx = tvm.context("cpu", 0)
+a = np.random.uniform(size=get_const_tuple(A.shape)).astype(dtype)
+b = np.random.uniform(size=get_const_tuple(B.shape)).astype(dtype)
+c = tvm.nd.array(np.zeros(get_const_tuple(C.shape), dtype=dtype), ctx)
+func(tvm.nd.array(a, ctx), tvm.nd.array(b, ctx), c)
+np.testing.assert_allclose(c.asnumpy(), np.dot(a, b.T), rtol=1e-3)
+
+######################################################################
+# We compare the tensorize version with that :code:`numpy.dot` produces,
+# ensure our implementation is correct.
+#
+# Reduce-update for Tensorize
+# ------------------------------------
+# Let's then move one step forward.
+# Assume our accelerator could only multiply a vector by a square matrix,
+# in which the vector size needs to be no larger than 16.
+# Given such hardware constrain, now we need to split the reduce axis as following,
+#
+zo, zi = s[C].split(z, factor=factor)
+s[C].reorder(x, yo, zo, yi, zi)
+
+######################################################################
+# However, since the tensorize intrinsic now only covers a part of the reduce axis,
+# instead of using one "body" function, TVM requires a :code:`reduce_reset` function,
+# which will be invoked before the reduce for-loop, and a :code:`reduce_update` function,
+# which defines the "update" computing strategy.
```

Comment:
Sorry @ZihengJiang I deleted your comment by mistake (I was trying to delete mine)
Regarding the difference between update and body, I explained here: https://github.com/dmlc/tvm/pull/1774/files#diff-baa4de13ce6d932de43e0eb5c4cb8f16R254
Let me know if you think it can be described more clearly.

Reply:
LGTM, thanks for the explaination :)